src/read.c

```diff
@@ -3746,19 +3780,32 @@ avifResult avifDecoderReset(avifDecoder * decoder)
                                                 decoder->imageDimensionLimit,
                                                 data->diag),
                           AVIF_RESULT_INVALID_IMAGE_GRID);
+            colorCodecType = avifDecoderItemGetGridCodecType(colorItem);
+            AVIF_CHECKERR(colorCodecType != AVIF_CODEC_TYPE_UNKNOWN, AVIF_RESULT_INVALID_IMAGE_GRID);
```

Comment:
can you please change this to a normal if condition?

Use of `AVIF_CHECKERR` makes this a double negation and difficult to parse.

here and elsewhere. i have marked some of the places which i found.

Reply:
Yes, if it was an assertion i would certainly prefer the latter. But i don't think the analogy you have here is right. The AVIF_CHECKERR statement when expanded looks like this:

```c
if (!(colorCodecType != AVIF_CODEC_TYPE_UNKNOWN)) {
    return AVIF_FALSE;
}
```

And if we were to write it without a macro, we certainly wouldn't write it like this. :)

Reply:
The same could be said about `assert()` if it was implemented as:
```c
#define assert(x) \
if (!(x)) {
    abort();
}
```

And I do read `AVIF_CHECKERR()` as "check this and continue if true", which is very similar to the assertion pattern. Anyway.